### PR TITLE
remove monitoring and add gcsdumpxml

### PR DESCRIPTION
**Pull request by iakinyemi**
Misc changes, include removing monitoring command line options and adding XML config dumper

### CODE REVIEW

**Comment by jiwaylee** [comment]
I am here.

File: bld.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 # bld.sh - build the programs on Linux.
 
-source ~/lbm.sh
+source ../../lbm.sh
```
Comment (jiwaylee) [inline bld.sh]: changing back to original path

Comment (fordsfords) [inline bld.sh]: Ah, I checked in that change and didn't tell anybody. No problem with changing it back. I'll re-visit in the future.

File: gcsdumpxml.c [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+
+ 
+
+int rcv_callback(lbm_rcv_t *rcv, lbm_msg_t *msg, void *clientd){
+ printf("\n Receiver callback placeholder");
+ return 0;
+}
+
+
+const char purpose[] = "Purpose: Receive messages on a single topic.";
```
Comment (fordsfords) [inline gcsdumpxml.c]: I don't think the purpose string is correct?

File: gcsusrc.c [hunk 3]
```diff
@@ -1326,7 +1216,8 @@ int main(int argc, char **argv)
 			} else {
 				sprintf(message, "message %lu", count);
 			}
-			exinfo.ume_msg_clientd = (void *)((lbm_uint_t)count + 1);
+			exinfo.ume_msg_clientd = (void *)((long long)count + 1);
```
Comment (fordsfords) [inline gcsusrc.c:1219]: 1. Was this change a fix to a bug?
2. I've never liked commenting out old code. Git does a fine job of showing you older code.
3. One of the C compilers, I don't remember which one, doesn't like double-slash comments for standard C.

File: bld.sh (hunk 1 above)
Comment (fordsfords) [reply bld.sh]: Oh, and it should probably be "../lbm.sh". Omit the extra "../".

**Comment by fordsfords** [comment]
I can't leave a comment on the binary files, so I'll mention that the file "gcs_tools-solution.backup.tz" should be renamed on top of "gcs_tools-solution-backup.tz" (change a dot to a dash). I corrected the wiki page that incorrectly named the file.

Also, I think the file "gcs_tools-solution.zip" should not be in there. Either that or the wiki instructions should be updated.

File: gcsmrcv.c [hunk 4]
```diff
@@ -691,7 +613,7 @@ lbm_ulong_t get_loss_or_print_stats(int num_ctxs, lbm_context_t * ctxs[], int pr
 				print_stats(stdout, stats[nstat]);
 			}
 			else{
-				/* Accumlate transport level loss */
+				/* Accumulate transport level loss */
```
Comment (fordsfords) [inline gcsmrcv.c:616]: How do you catch these typos??? :-)

File: monitoring_notes.txt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#                 (only if event queues are being used)
+receiver    monitor_interval 600
+#                 (warning, may not be appropriate for applications that create large
+#                  numbers of receivers; only recommended for UM version 6.14
+#                  and beyond and "context monitor_format pb" is used.)
+wildcard_receiver monitor_interval 600
+#                 (warning, may not be appropriate for applications that create large
+#                  numbers of wildcard receivers; only recommended for UM version 6.14
+#                  and beyond and "context monitor_format pb" is used.)
+context     monitor_transport_opts context|default_interface="en0"; context|resolver_unicast_daemon="en0"->10.1.2.3:12000;source|transport=tcp
```
Comment (fordsfords) [inline monitoring_notes.txt:67]: I think there needs to be some explanation here about the lbmrd. It's not obvious that the intent is to have a parallel unicast-only TRD for monitoring, separate from their application TRD. I was hoping the doc described it, but it doesn't do it well. This gets closer: https://github.com/UltraMessaging/mon_demo#configuration-goals

**Review by fordsfords — CHANGES_REQUESTED** [review]
There are just a few things that should be changed before merging. See comments.

File: gcsdumpxml.c (hunk 2 above)
Comment (iakinyemi) [reply gcsdumpxml.c]: I took it out since it wasn't used, thanks.

File: gcsusrc.c (hunk 3 above)
Comment (iakinyemi) [reply gcsusrc.c:1219]: This was to get rid of a compiler warning. I took out the commented-out old code.

File: gcsmrcv.c (hunk 4 above)
Comment (iakinyemi) [reply gcsmrcv.c:616]: dumb luck :)

> Either that or the wiki instructions should be updated.

I removed the folder, included by mistake.

**Pull request merged**